base_tier_validation/wizard/comment_wizard.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CommentWizard(models.TransientModel):
+    _name = 'comment.wizard'
+    _description = 'Comment Wizard'
+
+    validate_reject = fields.Char()
+    res_model = fields.Char()
+    view_id = fields.Many2one(
+        comodel_name='ir.ui.view',
+    )
+    definition_ids = fields.Many2many(
+        comodel_name='tier.definition',
+    )
+    comment = fields.Char(
+        required=True,
+    )
+
+    @api.multi
+    def add_comment(self):
+        self.ensure_one()
+        action = self.env['ir.model.data'].search([
+            ('model', '=', self.res_model),
+            ('res_id', '=', self.view_id.id),
```

Comment:
I am a bit confused by this `view_id` variable you are using.
Are we trying to get the action object? Why is it related to view?

Reply:
If I am correct that you try to get the inheriting object, i.e., purchase.request, purchase.order, etc.
Naming variable using `action` is very misleading. You better use, i.e., object or record...

Reply:
I will change this code follow https://github.com/OCA/server-ux/pull/98#discussion_r341811986